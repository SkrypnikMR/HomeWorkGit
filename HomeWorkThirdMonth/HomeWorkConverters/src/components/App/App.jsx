import React from 'react';
import { StAppDiv } from './styled';
import Header from '../Header';

const App = () => {
    return (
        <StAppDiv>
            <Header />
        </StAppDiv>
    );
};

export default App;
