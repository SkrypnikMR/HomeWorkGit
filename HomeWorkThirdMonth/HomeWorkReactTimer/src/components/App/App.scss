.app-dark, .app-light{
    font-family: 'Play', sans-serif;
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    flex-direction: column;
}
.app-light{
    background-color: rgb(229, 208, 237);
}
.app-dark{
    background-color: rgb(91, 88, 92);
}
