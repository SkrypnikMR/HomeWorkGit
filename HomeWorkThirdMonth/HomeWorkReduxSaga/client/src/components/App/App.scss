body {
    font-family: 'Play', sans-serif;
    margin: 0 auto;
   	background-color: #1B264F;
}
.app {
    display: flex;
    flex-direction: column;
    height: 100vh; 
}