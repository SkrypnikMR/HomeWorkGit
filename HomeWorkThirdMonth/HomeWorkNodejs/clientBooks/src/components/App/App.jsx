import React, { Component } from 'react';
import Header from '../Header';
import Content from '../Content';
import './App.scss';

import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
class App extends Component {
    constructor() {
        super();
    }
    render() {
        return (
            <div className="app">
                <Router>
                    <Header />
                    <Switch>
                        <Route exact path='/'><Content/></Route>
                    </Switch>
                </Router>
            </div>
        )
    }
}

export default App;

