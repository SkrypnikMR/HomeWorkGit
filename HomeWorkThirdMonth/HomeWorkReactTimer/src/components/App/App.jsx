import React, { Component } from 'react';
import Header from '../Header';
import Main from '../Main';
import Footer from '../Footer';
import { ThemeContext, themes } from '../../context/themeContext';
import './App.scss';
import { icon } from '../../helper/iconCreator';

class App extends Component {
    constructor(props) {
        super(props);
        this.state = { theme: themes.light, bulb: icon(themes.light) };
    }

    toggleTheme() {
        const { theme } = this.state;
        if (theme === themes.light) return this.setState({ theme: themes.dark, bulb: icon(themes.dark) });
        this.setState({ theme: themes.light, bulb: icon(themes.light) });
    }

    render() {
        const { theme, bulb } = this.state;
        return (
            <ThemeContext.Provider value={{ toggleTheme: this.toggleTheme.bind(this), bulb }}>
                <div className={`app-${theme}`}>
                    <Header />
                    <Main />
                    <Footer />
                </div>
            </ThemeContext.Provider >
        );
    }
}

export default App;
