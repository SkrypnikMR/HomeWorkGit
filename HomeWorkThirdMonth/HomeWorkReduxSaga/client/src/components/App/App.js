import React from 'react';
import Header from '../Header';
import Movies from '../Movies';
import './App.scss';
import Footer from '../Footer';
import { NotificationContainer } from 'react-notifications';
import 'react-notifications/lib/notifications.css';

const App = () => (
    <div className='app'>
        <Header />
        <Movies />
        <Footer />
        <NotificationContainer />
    </div>
);

export default App;
