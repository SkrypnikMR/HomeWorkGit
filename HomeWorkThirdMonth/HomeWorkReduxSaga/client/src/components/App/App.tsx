import React from 'react';
import Header from '../Header';
import Movies from '../Movies';
import './App.scss';
import Footer from '../Footer';
import { NotificationContainer } from 'react-notifications';
import 'react-notifications/lib/notifications.css';
import { useTheme } from '../hoc/withTheme.jsx';

const App = () => {
    const { theme } = useTheme();
    return (
        <div className={theme ? 'app' : 'app__dark'}>
            <Header />
            <Movies />
            <Footer />
            <NotificationContainer />
        </div>
    );
}



export default App;
