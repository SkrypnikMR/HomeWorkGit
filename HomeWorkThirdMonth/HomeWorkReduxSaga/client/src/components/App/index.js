import App from './App';
import { withTheme } from '../hoc/withTheme.jsx';

export default withTheme(App);
