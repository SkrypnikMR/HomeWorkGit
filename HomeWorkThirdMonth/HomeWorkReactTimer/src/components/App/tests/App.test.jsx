import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../App';

describe('App', () => {
    it('Should match snapshot', () => {
        const component = shallow(<App />);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should have App ', () => {
        const component = mount(<App />);
        expect(component.find('App')).toHaveLength(1);
    });
    it('Should have Header ', () => {
        const component = mount(<App />);
        expect(component.find('Header')).toHaveLength(1);
    });
});
