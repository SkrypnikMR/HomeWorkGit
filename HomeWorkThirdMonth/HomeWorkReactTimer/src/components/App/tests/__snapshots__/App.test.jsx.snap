// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `"<div class=\\"app-light\\"><header class=\\"header\\"><h1>React Timer</h1><button>â˜€</button></header><div class=\\"main\\"><div class=\\"timer\\"><p class=\\"timer__tablo\\">00:00:00</p><div class=\\"timer__controll-set\\"><input type=\\"text\\" placeholder=\\"Enter time for Timer in sec\\" value=\\"\\"/><button disabled=\\"\\">Set</button></div><div class=\\"timer__controll-operations\\"><button disabled=\\"\\">Stop</button><button disabled=\\"\\">Start</button><button disabled=\\"\\">Reset</button></div><div class=\\"notification-container notification-container-empty\\"><div></div></div></div></div><header class=\\"footer\\"><h3>React Timer made by:  <a href=\\"https://t.me/Skripnik2201\\" target=\\"_blank\\">@Skripnik2201</a></h3></header></div>"`;
